### ISSUE TITLE
typing.NamedTuple: no value for argument 'typename'

### ISSUE THREAD

**Issue by quodlibetor** [PyCQA/pylint#1418]
typing.NamedTuple doesn't seem to be handled correctly with the new-style metaclass:

### Steps to reproduce
```
$ cat nt.py
```
```python
from typing import NamedTuple

class Example(NamedTuple):
    mything: int

Example(1)  # this works

ex = Example(mything=1)  # python happy, pylint error
print(ex)
print(ex._asdict())  # python happy, pylint error
```
```
$ python nt.py
Example(mything=1)
OrderedDict([('mything', 1)])
```

### Current behavior
```
$ pylint -E nt.py
No config file found, using default configuration
************* Module nt
E:  8, 5: No value for argument 'typename' in constructor call (no-value-for-parameter)
E: 10, 6: Instance of 'Example' has no '_asdict' member (no-member)
```

### Expected behavior
no error

### pylint --version output
```
$ pylint --version
No config file found, using default configuration
pylint 1.7.0,
astroid 1.5.1
Python 3.6.1 (default, Apr 13 2017, 11:50:19)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

**Comment by PCManticore** [comment]
Fixed by https://github.com/PyCQA/astroid/commit/7eb703d2e2bec5ffb2e679c61ca0ea26d1413e27

**Comment by quodlibetor** [comment]
Awesome, thanks!

**Comment by wbkdef** [comment]
I'm still getting this issue with pylint 1.8.2.

`
import typing as tp
from typing import Callable, NamedTuple

class LineGuideInfo(tp.NamedTuple):
    line_num_criterion: Callable[[int], bool]
    visual_guide: str
    exclusive: bool

line_guide_header = LineGuideInfo(line_num_criterion=lambda x: x == 0, visual_guide="\n".join(["=" * 100] * 2), exclusive=True)
`

pylint output:

> C:\Users\wbruc\Desktop\git_repos\organization>pylint -dw -dc -dr C:\Users\wbruc\Desktop\git_repos\organization\bin_python_scripts\DEBUG_PyLint_Error_py_lint_filterer.py
> Using config file C:\Users\wbruc\Desktop\git_repos\organization\.pylintrc
> ************* Module DEBUG_PyLint_Error_py_lint_filterer
> C:\Users\wbruc\Desktop\git_repos\organization\bin_python_scripts\DEBUG_PyLint_Error_py_lint_filterer.py:9: [E1120(no-value-for-parameter), ] No value for argument 'typename' in constructor call


* NOTE - The error goes away when I don't "import typing as tp":

`
from typing import Callable, NamedTuple

class LineGuideInfo(NamedTuple):
    line_num_criterion: Callable[[int], bool]
    visual_guide: str
    exclusive: bool

line_guide_header = LineGuideInfo(line_num_criterion=lambda x: x == 0, visual_guide="\n".join(["=" * 100] * 2), exclusive=True)
`

**Comment by terrisgit** [comment]
I don't "import typing as tp" and it happens to me on pylint 1.8.2

**Comment by samuela** [comment]
I'm seeing the same thing now on pylint 2.3.1. I'm also not importing with anything like `import typing as tp`. In my code,
```python
from typing import NamedTuple
import jax.numpy as jp

Distribution = NamedTuple

class MVN(Distribution):
  loc: jp.array
  cov_cholesky: jp.array

# in another file...
true_dist = MVN(loc=jp.zeros((2, )),
                cov_cholesky=eigenvectors @ jp.diag(jp.sqrt(eigenvalues)))
```
which gives me exactly this error.

**Comment by macdjord** [comment]
@PCManticore Can you please reopen this issue, because the bug is not, in fact, solved.

**Comment by PCManticore** [comment]
Thanks for reproduction cases, folks, I can also reproduce the issue with some of them.

**Comment by hippo91** [comment]
@macdjord @samuela it seems that this bug does not occur anymore with latest version of `astroid` and `pylint`.
For example linting:
```
#pylint: disable=line-too-long, missing-module-docstring, missing-class-docstring
import typing as tp
from typing import Callable, NamedTuple

class LineGuideInfo(tp.NamedTuple):
    line_num_criterion: Callable[[int], bool]
    visual_guide: str
    exclusive: bool

line_guide_header = LineGuideInfo(line_num_criterion=lambda x: x == 0, visual_guide="\n".join(["=" * 100] * 2), exclusive=True)
```
gives
```
************* Module bug_pylint_1418_b
bug_pylint_1418_b.py:3:0: W0611: Unused NamedTuple imported from typing (unused-import)

------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 8.57/10, +0.00)
```
which seems to be correct.

Linting:
```
#pylint: disable=missing-module-docstring, missing-class-docstring
from typing import NamedTuple
import jax.numpy as jp

Distribution = NamedTuple

class MVN(Distribution):
    loc: jp.array
    cov_cholesky: jp.array

# in another file...
true_dist = MVN(loc=jp.zeros((2, )),
                cov_cholesky=eigenvectors @ jp.diag(jp.sqrt(eigenvalues)))
```
gives:
```
************* Module bug_pylint_1418
bug_pylint_1418.py:13:29: E0602: Undefined variable 'eigenvectors' (undefined-variable)
bug_pylint_1418.py:13:60: E0602: Undefined variable 'eigenvalues' (undefined-variable)

----------------------------------------------------------------------
Your code has been rated at -4.29/10 (previous run: -14.29/10, +10.00)
```
which also seems correct.

```
pylint --version
pylint 2.8.0-dev1
astroid 2.6.0-dev1
Python 3.9.1 (default, Dec 30 2020, 11:18:33) 
[GCC 7.5.0]
```
